integration/oasis/stationgraph/visited_map.go

```diff
@@ -0,0 +1,139 @@
+package stationgraph
+
+import "github.com/golang/glog"
+
+type visitedNodeInfo struct {
+	prevNodeID nodeID
+	pqElem     *pqElement
+	minCost    float64
+	minDist    float64
+	settled    bool
+}
+
+type visitedMap struct {
```

Comment:
Actually I don't very understand why name it `visitedmap`. Yes, it stores infos that have already visited, but it's not a really `map`, it walks on a `map` but only stores some discrete `nodes/edges`. I more like the name `queryHeap` that used in OSRM since it core feature is the `priorityQueue`, and the `priorityQueue` is based on heap(binary heap, d_ary heap, etc).

Reply:
`visitedMap` is the name be used in several generations' of our navigation engine, its already into the soul of me :)

`queryHeap` is a good name here, it is my purpose of this class, a container provides heap functionality(heap/pop/decrease) and also provide functionality like hashtable to check whether need to decrease/settled.  I will change it to `queryheap`.

Reply:
I know. Actually it's still difficult to me to imagine what it is when I see `visitedmap` until now, even I have worked on the codes for years, that why I cannot say I like it :).       
The purpose is same with what `queryHeap` does.       
Also, which one will you prefer between `cost` and `weight`? We'd better to unify one to use widely. Both of them look good to me, but I may prefer to use `weight` to unify with OSRM.

Reply:
Haha, frankly speaking, queryheap is better name :)  Let's try to end visitedmap from now.

Let's put `weight` for all the places, will change all `cost` to `weight`.